### PR TITLE
Fix integration tests

### PR DESCRIPTION
**Pull request by Erin-Mounts**

### CODE REVIEW

File: BridgeSDKTests/SBBParticipantManagerIntegrationTests.m [hunk 1]
```diff
@@ -284,10 +287,50 @@ - (void)testSetSharingScope
...
+    [self waitForExpectationsWithTimeout:5.0 handler:^(NSError *error) {
+        if (error) {
+            NSLog(@"Timeout creating external identifier %@: %@", externalIdentifier, error);
+            externalIdentifier = nil;
+        }
+    }];
+
+    if (!externalIdentifier) {
+        // If we couldn't create the externalIdentifier, just fail & bail (we've already logged it)
+        XCTAssert(NO);
```
Comment (syoung-smallwisdom) [inline BridgeSDKTests/SBBParticipantManagerIntegrationTests.m]: Does Obj-c have `XCTFail(@"Impossible code path");` ? It's always nice to be able to look at the travis logs and see a message for why it failed. Travis doesn't tell you *where* it failed.

File: BridgeSDKTests/SBBParticipantManagerIntegrationTests.m [hunk 2]
```diff
@@ -284,10 +287,50 @@ - (void)testSetSharingScope
...
+        return;
+    }
+    
     XCTestExpectation *expectSetIdentifier = [self expectationWithDescription:@"set external identifier"];
-    NSString *externalIdentifier = @"test-external-identifier";
     [SBBComponent(SBBParticipantManager) setExternalIdentifier:externalIdentifier completion:^(id  _Nullable responseObject, NSError * _Nullable error) {
-        XCTAssert(!error, @"Server accepted external identifier");
+        XCTAssert(!error, @"Server rejected external identifier %@", externalIdentifier);
```
Comment (syoung-smallwisdom) [inline BridgeSDKTests/SBBParticipantManagerIntegrationTests.m]: `XCTAssertNil()` ? Is that macro available?

File: BridgeSDKTests/SBBParticipantManagerIntegrationTests.m [hunk 3]
```diff
@@ -323,6 +367,33 @@ - (void)testSetExternalIdentifier
...
+        NSString *endpoint = [NSString stringWithFormat:@"%@/%@", DEV_EXTERNAL_ID_API, externalIdentifier];
+        [bridgeMan delete:endpoint headers:headers parameters:externalIdObject completion:^(NSURLSessionTask *task, id responseObject, NSError *error) {
+            if (error) {
+                NSLog(@"Failed to delete external ID %@\nError:%@\nResponse:%@", externalIdentifier, error, responseObject);
+                externalIdentifier = nil;
+            } else {
+                NSLog(@"Deleted test external ID %@", externalIdentifier);
+            }
+        }];
+    }];
```
Comment (syoung-smallwisdom) [inline BridgeSDKTests/SBBParticipantManagerIntegrationTests.m]: Should probably throw an assert if this fails, too. Just so you know that something went wrong.

File: BridgeSDKTests/SBBParticipantManagerIntegrationTests.m [hunk 4]
```diff
@@ -284,10 +287,50 @@ - (void)testSetSharingScope
...
+            NSLog(@"Created test external ID %@", externalIdentifier);
+        }
+        
+        [expectCreatedIdentifier fulfill];
+    }];
+    
+    [self waitForExpectationsWithTimeout:5.0 handler:^(NSError *error) {
+        if (error) {
+            NSLog(@"Timeout creating external identifier %@: %@", externalIdentifier, error);
+            externalIdentifier = nil;
```
Comment (syoung-smallwisdom) [inline BridgeSDKTests/SBBParticipantManagerIntegrationTests.m]: I think how expectations work has changed and you need to include an assert here if there was an error.

File: BridgeSDKTests/SBBParticipantManagerIntegrationTests.m [hunk 5]
```diff
@@ -284,10 +287,50 @@ - (void)testSetSharingScope
...
+      @"substudyId": substudyId,
+      @"type": @"ExternalIdentifier"
+      };
+    
+    XCTestExpectation *expectCreatedIdentifier = [self expectationWithDescription:@"created external identifier"];
+    id<SBBBridgeNetworkManagerProtocol> bridgeMan = SBBComponent(SBBBridgeNetworkManager);
+    [bridgeMan post:DEV_EXTERNAL_ID_API headers:headers parameters:externalIdObject completion:^(NSURLSessionTask *task, id responseObject, NSError *error) {
+        if (error) {
+            NSLog(@"Failed to create external ID %@\nError:%@\nResponse:%@", externalIdentifier, error, responseObject);
+            externalIdentifier = nil;
```
Comment (syoung-smallwisdom) [inline BridgeSDKTests/SBBParticipantManagerIntegrationTests.m]: Why log rather than assert?

Comment (Erin-Mounts) [reply BridgeSDKTests/SBBParticipantManagerIntegrationTests.m]: good point

File: BridgeSDKTests/SBBParticipantManagerIntegrationTests.m (hunk 4 above)
Comment (Erin-Mounts) [reply BridgeSDKTests/SBBParticipantManagerIntegrationTests.m]: 👍

Comment (Erin-Mounts) [reply BridgeSDKTests/SBBParticipantManagerIntegrationTests.m]: ...actually remembered the reason I didn't do this: I want to make sure the test run doesn't halt prematurely so as much as possible of the test accounts etc. that it sets up in the test study can get cleaned up after each run even when there are failures. I still go in to the test study periodically to clean up stuff that got left behind when it hits an assert in the Bridge or iOS SDK, but it gets out of hand quickly if I don't let these continue after errors.

File: BridgeSDKTests/SBBParticipantManagerIntegrationTests.m (hunk 5 above)
Comment (Erin-Mounts) [reply BridgeSDKTests/SBBParticipantManagerIntegrationTests.m]: see other comment about log vs assert

**Pull request merged**